### ISSUE TITLE
Auto-Type Window title with | pipe symbol is interpreted as logical OR

### ISSUE THREAD

**Issue by stefan123t** [keepassxreboot/keepassxc#6413]
## Overview
I have an old database coming from Strip, Keepass 1.x, 2.x, XKeepass and now Keepass XC.
I upgraded to KeepassXC 2.7.0 snapshot as part of a pull request by hifi.
When hitting autotype in the latest KeepassXC the Window Title with `|` pipe symbol is interpreted as OR.
I have many Window Titles configured which used to interpret the pipe symbol as a literal.
I believe some Browser used to have windows titles like: `<page title> | <browser name>`
In order to catch them all I have a lot of entries: `<page title> | *`
But also: `<page title> - *`

Now with KeepassXC 2.7.0 snapshot it will interpret this as an OR.
And many of the Auto-type entries come up because they match a single `*`

## Steps to Reproduce
1. Create two password entries, each with auto-type `<page A/B> | *`
2. Use Auto-type on `page A`, e.g. paypal
3. The Auto-type entry for `page B` will also show up as it does not match page B but `*` after the pipe symbol

## Expected Behavior
The Auto-Type entries should only match a literal `|` pipe symbol and not a logical `OR`.
Please allow a way to escape pipe symbols in Window title e.g. `\|` or better add some Regex feature to interpret `\(page A\|page B\)`.
This could be enclosed in curly braces `{OR}`, `{REGEX:(page A|page B)}` in order to better align with the ubiquituous syntax of Keepass DBs.

I can also simply remove the `|` pipe symbol and - dash symbols from my Window Titles.
This saves me several auto-type entries per Password.
I just kept them to have a clear separator between Page Title and Browser Name.
In order to have a clear distinction between: `EBay | Browser Name` and `EBay Login | Browser Name` etc.

As mentioned above it would be super-cool to have RegEx for character classes or other things:
`EBay Login {REGEX:[|-]} *`
This would require a dash or pipe symbol after Login and before any Browser name

## Actual Behavior
Both entries will match for any page.

## Context
This may be a regression or a new feature which I may have missed.
Especially #6268 or #6296 might be related.

KeePassXC - Version 2.7.0-snapshot
Build Type: Snapshot
Revision: ee9d0b5

Operating system: Linux Mint 20.1
CPU architecture: x86_64
Kernel: linux 5.8.0-48-generic

**Comment by droidmonkey** [comment]
I didn't touch window title matching code??

**Comment by stefan123t** [comment]
Dear @droidmonkey,

Thanks for looking into this. I have changed all my Window titles with pipes in them = fixed for me. But it might be a regression introduced somewhere over time.

I am coming from the version being installed on Linux Mint 21 as documented in issue 324, ie before testing the 2.7 snapshot with hifi's pull request. This was a stark contrast to Xkeepass and Keepas 1/2 and I believe I have not seen it before in KeepassXC either:

KeePassXC - Version 2.4.3 Revision: 5d6ef0c
Qt 5.12.8 Debugging mode is disabled.
Operating system: Linux Mint 20.1

This version is not having the new selection dialog. Maybe the two revisions can be compared when looking at it from the right angle? You guys seem to have the right perspective on all this fine code =^)

Kind regards,
Stefan

**Comment by droidmonkey** [comment]
Yup this is my fault. I converted our window title matching to use the same conversion function as the search bar. So wildcards (*, ?, |) all worked as regular expressions. I do note there is a bug here in that you cannot escape the pipe symbol. I am working to fix that now.

**Comment by stefan123t** [comment]
Dear Jonathan, as mentioned in 6268 this could be an appreciated feature.
I just noticed the difference between the previous KeepassXC and the 2.7 version.
So we are not following the Keepass 1/2 "standard" here, which is not a problem for me, as I worked around it.
But if you can come up with an advanced window title matching code supporting some regex or else here, I would appreciate it too. Best regards, Stefan

**Comment by stefan123t** [comment]
@droidmonkey, should this be tagged with PRE-RELEASE BUG as other 2.7.0 issues ?

**Comment by obambrough** [comment]
(Running KeyPassXC under MacOS) I just updated from KeyPassXC 2.6.4 to 2.6.6 and believe this issue is present in 2.6.6.

**Comment by droidmonkey** [comment]
No it certainly is not present in 2.6.6, this is only impacting develop branch.

**Comment by obambrough** [comment]
I must have been mistaken. I tried to reproduce with 2.6.6 but couldn't. I upgraded to 2.6.6 this morning but then auto-type wasn't finding my existing entry. I remember seeing something odd - I re-added the URL to the auto-type list for the specific entry (by selecting the window title from the drop-down list) and noticed it was identical to the existing entry but there was a loud-speaker icon at the end of the URL. I thought maybe it was telling me it was a duplicate entry or some such. Unfortunately I can't reproduce the behaviour.
